sdk/src/main/scala/com/horizen/account/state/AccountState.scala

```diff
@@ -111,14 +116,52 @@ class AccountState(val params: NetworkParams,
       stateView.applyMainchainBlockReferenceData(mcBlockRefData).get
     }
 
-    for (tx <- mod.sidechainTransactions) {
-      stateView.applyTransaction(tx).get
+    // get also list of receipts, useful for computing the receiptRoot hash
+    val receiptList = new ListBuffer[EthereumReceipt]()
+    val blockNumber = stateView.getHeight
+    val blockHash = idToBytes(mod.id)
+    var cumGasUsed : BigInteger = BigInteger.ZERO
+
+    for ((tx, txIndex) <- mod.sidechainTransactions.zipWithIndex) {
+      stateView.applyTransaction(tx, cumGasUsed) match {
+        case Success(consensusDataReceipt) =>
+          val txGasUsed = consensusDataReceipt.cumulativeGasUsed.subtract(cumGasUsed)
+          // update cumulative gas used so far
+          cumGasUsed = consensusDataReceipt.cumulativeGasUsed
+          val ethTx = tx.asInstanceOf[EthereumTransaction]
+
+          val txHash = idToBytes(ethTx.id)
+
+          val contractAddress = if (ethTx.getTo != null && !stateView.isEoaAccount(ethTx.getTo.address()) ) {
+            ethTx.getTo.address()
+          } else {
+            new Array[Byte](Address.LENGTH)
```

Comment:
how can we get the real contract address?
maybe it is returned by EVM message processor in "returnData" bytes?

Reply:
After discussion and some research on docs and geth code I fixed it getting the contractAddress via w3j util function, equivalent to the createAddress() in geth (triggered also by CREATE opcode).
Note: geth has also a CREATE2 opcode which may be optionally used in a smart contract solidity implementation in order to deploy another contract with a predefined address, created with a different formula.